src/session_impl.cpp

```diff
@@ -5661,7 +5655,7 @@ namespace aux {
 		m_dht_settings = settings;
 	}
 
-	void session_impl::set_dht_state(dht::dht_state const& state)
+	void session_impl::set_dht_state(dht::dht_state state)
 	{
 		m_dht_state = state;
```

Comment:
hmm...I reasoned in the same way, but I was under the impression that in this particular case, the compiler can "cleverly" detect that there is no further use of the variable and only the destructor is called. I could be wrong in that assumption.